python/cudf/cudf/core/column/timedelta.py

```diff
@@ -202,6 +202,11 @@ def _binaryop(self, other: ColumnBinaryOperand, op: str) -> ColumnBase:
         elif other.dtype.kind in {"f", "i", "u"}:
             if op in {"__mul__", "__mod__", "__truediv__", "__floordiv__"}:
                 out_dtype = self.dtype
+            elif op in {"__eq__", "NULL_EQUALS"}:
```

Comment:
What about `__ne__`?

Generally pandas behavior for an invalid comparison between types is

1. Return array of `True`s for ne
2. Return array of `False`s for eq
3. TypeError for other ops

Reply:
Thanks @mroeschke for catching this. I got it handled now.